pkg/cluster/pod.go

```diff
@@ -399,11 +415,13 @@ func (c *Cluster) getPatroniMemberData(pod *v1.Pod) (patroni.MemberData, error)
 }
 
 func (c *Cluster) recreatePod(podName spec.NamespacedName) (*v1.Pod, error) {
-	ch := c.registerPodSubscriber(podName)
-	defer c.unregisterPodSubscriber(podName)
-	stopChan := make(chan struct{})
+	stopCh := make(chan struct{})
+	ch := c.registerUnbufferedPodSubscriber(podName)
+	// send special end event to trigger removal of PodEvent channel by processPodEvent
+	defer c.ReceivePodEvent(PodEvent{PodName: types.NamespacedName(podName), EventType: PodEventEnd})
```

Comment:
@FxKu a bit confusing with the function name, as it sends event, not receives.

Reply:
I also stumbled over the name. But maybe I can get around sending artificial podEvents, after all, see #1891